pandas/core/dtypes/inference.py

```diff
@@ -267,6 +267,26 @@ def is_list_like(obj):
             not isinstance(obj, string_and_binary_types))
 
 
+def is_array_like(obj):
+    """
+    Check if the object is array-like.
+
+    For an object to be considered array-like, it must be list-like and
+    have a `dtype` attribute.
+
+    Parameters
+    ----------
+    obj : The object to check.
+
+    Returns
+    -------
+    is_array_like : bool
+        Whether `obj` has array-like properties.
+    """
```

Comment:
can you show Index, Series, ndarray (and a scalar) as examples

Reply:
as a followup we could replace code in various places to use this

Reply:
Done.